medusa/show_queue.py

```diff
@@ -367,32 +369,51 @@ def run(self):
             # this usually only happens if they have an NFO in their show dir which gave us a Indexer ID that
             # has no proper english version of the show
             if getattr(s, 'seriesname', None) is None:
-                logger.log(u"Show in {} has no name on {}, probably searched with the wrong language.".format
+                logger.log(u"Show in {0} has no name on {1}, probably searched with the wrong language.".format
                            (self.showDir, indexerApi(self.indexer).name), logger.ERROR)
 
-                ui.notifications.error("Unable to add show", "Show in " + self.showDir + " has no name on " +
-                                       str(indexerApi(self.indexer).name) + ", probably the wrong language. "
-                                       "Delete .nfo and manually add the correct language.")
+                ui.notifications.error('Unable to add show Show',
+                                       'Show in {0} has no name on {1}, probably the wrong language. \
+                                       Delete .nfo and manually add the correct language.'
+                                       .format(self.showDir, indexerApi(self.indexer).name))
                 self._finishEarly()
                 return
             # if the show has no episodes/seasons
             if not s:
                 logger.log(u"Show " + str(s['seriesname']) + u" is on " + str(indexerApi(self.indexer).name) +
                            u" but contains no season/episode data.")
                 ui.notifications.error("Unable to add show",
-                                       "Show " + str(s['seriesname']) + " is on " + str(indexerApi(
-                                           self.indexer).name) + " but contains no season/episode data.")
+                                       "Show {0} is on {1} but contains no season/episode data.".
+                                       format(s['seriesname'], indexerApi(self.indexer).name))
                 self._finishEarly()
+
+                return
+
+            # Check if we can already find this show in our current showList.
+            try:
+                check_existing_shows(s, self.indexer)
+            except IndexerShowAllreadyInLibrary as e:
+                logger.log(u"Could not add the show %s, as it already is in your library."
+                           u" Error: %s" % (s['seriesname'], e.message), logger.WARNING)
+                ui.notifications.error(
+                    'Unable to add show',
+                    'reason: {0}' .format(e.message)
+                )
+                self._finishEarly()
+
+                # Clean up leftover if the newly created directory is empty.
+                delete_empty_folders(self.showDir)
```

Comment:
This will delete the "Shows folder" if there aren't any other shows in that folder.